components/brave_wallet/browser/brave_wallet_provider_impl.cc

```diff
@@ -548,6 +548,49 @@ void BraveWalletProviderImpl::GetEncryptionPublicKey(const std::string& address,
                      std::move(id), address));
 }
 
+void BraveWalletProviderImpl::Decrypt(
+    const std::string& untrusted_encrypted_data_json,
+    const std::string& address,
+    RequestCallback callback,
+    base::Value id) {
+  data_decoder::JsonSanitizer::Sanitize(
+      untrusted_encrypted_data_json,
+      base::BindOnce(&BraveWalletProviderImpl::ContinueDecryptWithSanitizedJson,
```

Comment:
cc @diracdeltas I think this is the main part that interests you.  Maybe also in the first commit with the Decrypt tweetnacl work.

Reply:
@kdenhartog